axonserver/src/main/java/io/axoniq/axonserver/message/command/CommandRegistrationCache.java

```diff
@@ -138,25 +145,36 @@ private boolean provides(ClientIdentification client, String cmd) {
      * @return a command handler for the command (or null of none found)
      */
     public CommandHandler getHandlerForCommand(String context, Command request, String routingKey) {
-        ConsistentHash hash = consistentHashPerContext.get(context);
-        if( hash == null) return null;
-        String client = hash
-                                                .getMember(routingKey, request.getName())
-                                                .map(ConsistentHash.ConsistentHashMember::getClient)
-                                                .orElse(null);
-        if( client == null ) return null;
+        String command = request.getName();
+        RoutingSelector<String> routingSelector = routingSelector(context, command);
+        String client = routingSelector.selectHandler(routingKey).orElse(null);
+        if (client == null) {
+            return null;
+        }
         return commandHandlersPerClientContext.get(new ClientIdentification(context, client));
```

Comment:
Nit: you could change this line to:
```
return routingSelector.selectHandler(routingKey)
        .orElse(commandHandlersPerClientContext.get(new ClientIdentification(context, client)));
```

Reply:
client is a String, we want here the `CommandHandler`

Reply:
Aaah fair point, so a `map` operation should be added I guess.

Reply:
Done